src/Symfony/Component/HttpKernel/Exception/FailedToResolveValueException.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\Exception;
+
+/**
+ * Can be thrown by any value resolver (a class that extends ValueResolverInterface)
+ * to specify why exactly it cannot resolve a value.
+ */
+class FailedToResolveValueException extends \RuntimeException
```

Comment:
WDYT of this?
```suggestion
 * Lets value resolvers tell when an argument could be under their watch but failed to be resolved.
 */
class NearMissValueResolverException extends \RuntimeException
```

Reply:
I don't mind it, it's just that it doesn't make a lot of sense to me that `ServiceValueResolver` throws an exception on pretty much everything that could not be fetched from the container. Based on this logic, it will be a "near miss" every time the service does not exist 😄
But if you prefer the suggested naming better, I can change it 👍

Reply:
Sure, I don't have a strong preference here :)